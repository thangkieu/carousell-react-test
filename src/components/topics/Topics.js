/**
 * @author Thang Kieu
 *
 * @description Topics component
 * list all of topics base on data from Topics Container
 */

import React from 'react';
import { Link } from 'react-router';

import Topic from '../topic';
import './topics.css';

const Topics = ({ topics, upVoteTopic, downVoteTopic, onRefresh }) => (
  <div className="topics-container">
    <div className="container">

      <div className="row">
        <div className="col-xs-12 col-sm-12 col-md-8 col-md-push-2">
          <div className="clearfix">
            <h2 className="pull-left">Topic list</h2>
            <Link to="/add" className="pull-right btn btn-primary btn-add-topic">Add Topic</Link>
            <button className="pull-right btn btn-link btn-refresh" onClick={onRefresh}>Refresh</button>
          </div>
          <div className="topic-list">
            {
              topics && topics.length > 0 &&
              topics.map(topic => (
                <Topic
                  key={topic.id}
                  topic={topic}
                  onUpVoteTopic={() => upVoteTopic && upVoteTopic(topic.id)}
                  onDownVoteTopic={() => downVoteTopic && downVoteTopic(topic.id)}
                />
              ))
            }
          </div>
        </div>
      </div>

    </div>
  </div>
);

export default Topics;
