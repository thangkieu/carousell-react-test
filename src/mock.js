export const topics = [{
  title: 'Why do people say "let love find you, don\'t look for it"?',
  content: '"If you sit home playing video games all day, you won\'t meet anyone." Heh. Heh heh heh. HEH HEH!I was 49 years old, and had never been married. In fact, I had barely dated as an adult. I\'d had one',
  id: 1,
  upvote: 10,
  downvote: 0
}, {
  title: 'What psychological tricks and hacks are useful to know?',
  content: 'Here are my 33 secret secrets which maybe I have never shared with anyone.',
  id: 2,
  upvote: 20,
  downvote: 4
}, {
  title: 'What should you never say to a programmer?',
  content: 'So Mark Zuckerberg came to Nigeria and a lot of people wouldn\'t shut up about it while he was around',
  id: 3,
  upvote: 13,
  downvote: 0
}, {
  title: 'What is a fun psychological trick to try on someone?',
  content: 'This one’s always funny… or at least it is when I use it on my friends.Say you\'re sitting next to your friend in a library. Your friend is studying intensely for the upcoming exams',
  id: 4,
  upvote: 19,
  downvote: 0
}, {
  title: 'Which is the better platform, Squarespace or GoDaddy?',
  content: 'Lorem',
  id: 5,
  upvote: 10,
  downvote: 2
}, {
  title: 'What is the best advice you can give to a junior programmer?',
  content: 'Oh I have a bunch of these.Been a Software Developer in multiple companies for close to 10 years and I could not stress more on the following',
  id: 6,
  upvote: 4,
  downvote: 3
}, {
  title: 'How good of a programmer is Mark Zuckerberg?',
  content: 'In 2006, Zuckerberg gave up coding to focus on running the business of Facebook',
  id: 7,
  upvote: 15,
  downvote: 5
}, {
  title: 'What are the best comments in a code?',
  content: 'Found at the beginning of an infinite loop',
  id: 8,
  upvote: 12,
  downvote: 2
}, {
  title: 'What are some things that look easy but are difficult?',
  content: 'Doing stand up comedy. I do it as an amateur for fun.What I thought it was before I did it: You walk up, tell jokes for 10 minutes, then collect your free beer or $20.',
  id: 9,
  upvote: 12,
  downvote: 3
}, {
  title: 'How India\'s Aadhaar database is maintained? What are the data structures involved? How search is performed (from an algorithmic perspective)?',
  content: '',
  id: 10,
  upvote: 10,
  downvote: 4
}];
