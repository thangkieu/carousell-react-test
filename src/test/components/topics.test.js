import React from 'react';
import Topics from '../../components/topics';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { topics as TOPICS } from '../../mock';

it('Should match snapshot with full data', () => {
  const component = renderer.create(<Topics topics={TOPICS} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Functions pass to component should be called', () => {
  const onRefresh =jest.fn();
  const component = mount(<Topics topics={TOPICS} onRefresh={onRefresh} />);

  component.find('.btn-refresh').simulate('click');
  expect(onRefresh).toBeCalled();
});
