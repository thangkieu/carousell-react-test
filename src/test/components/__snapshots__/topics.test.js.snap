// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot with full data 1`] = `
<div
  className="topics-container"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-sm-12 col-md-8 col-md-push-2"
      >
        <div
          className="clearfix"
        >
          <h2
            className="pull-left"
          >
            Topic list
          </h2>
          <a
            className="pull-right btn btn-primary btn-add-topic"
            onClick={[Function]}
            style={Object {}}
          >
            Add Topic
          </a>
          <button
            className="pull-right btn btn-link btn-refresh"
            onClick={undefined}
          >
            Refresh
          </button>
        </div>
        <div
          className="topic-list"
        >
          <div
            className="topic-item"
          >
            <h4>
              Why do people say "let love find you, don't look for it"?
            </h4>
            <p>
              "If you sit home playing video games all day, you won't meet anyone." Heh. Heh heh heh. HEH HEH!I was 49 years old, and had never been married. In fact, I had barely dated as an adult. I'd had one
            </p>
            <button
              className="btn btn-default btn-sm btn-vote"
              onClick={[Function]}
            >
              <span>
                10
                 | 
              </span>
              <span>
                Vote
              </span>
            </button>
            <button
              className="btn btn-link btn-sm btn-downvote"
              onClick={[Function]}
            >
              <span>
                DownVote
              </span>
            </button>
          </div>
          <div
            className="topic-item"
          >
            <h4>
              What psychological tricks and hacks are useful to know?
            </h4>
            <p>
              Here are my 33 secret secrets which maybe I have never shared with anyone.
            </p>
            <button
              className="btn btn-default btn-sm btn-vote"
              onClick={[Function]}
            >
              <span>
                20
                 | 
              </span>
              <span>
                Vote
              </span>
            </button>
            <button
              className="btn btn-link btn-sm btn-downvote"
              onClick={[Function]}
            >
              <span>
                4
                 | 
              </span>
              <span>
                DownVote
              </span>
            </button>
          </div>
          <div
            className="topic-item"
          >
            <h4>
              What should you never say to a programmer?
            </h4>
            <p>
              So Mark Zuckerberg came to Nigeria and a lot of people wouldn't shut up about it while he was around
            </p>
            <button
              className="btn btn-default btn-sm btn-vote"
              onClick={[Function]}
            >
              <span>
                13
                 | 
              </span>
              <span>
                Vote
              </span>
            </button>
            <button
              className="btn btn-link btn-sm btn-downvote"
              onClick={[Function]}
            >
              <span>
                DownVote
              </span>
            </button>
          </div>
          <div
            className="topic-item"
          >
            <h4>
              What is a fun psychological trick to try on someone?
            </h4>
            <p>
              This one’s always funny… or at least it is when I use it on my friends.Say you're sitting next to your friend in a library. Your friend is studying intensely for the upcoming exams
            </p>
            <button
              className="btn btn-default btn-sm btn-vote"
              onClick={[Function]}
            >
              <span>
                19
                 | 
              </span>
              <span>
                Vote
              </span>
            </button>
            <button
              className="btn btn-link btn-sm btn-downvote"
              onClick={[Function]}
            >
              <span>
                DownVote
              </span>
            </button>
          </div>
          <div
            className="topic-item"
          >
            <h4>
              Which is the better platform, Squarespace or GoDaddy?
            </h4>
            <p>
              Lorem
            </p>
            <button
              className="btn btn-default btn-sm btn-vote"
              onClick={[Function]}
            >
              <span>
                10
                 | 
              </span>
              <span>
                Vote
              </span>
            </button>
            <button
              className="btn btn-link btn-sm btn-downvote"
              onClick={[Function]}
            >
              <span>
                2
                 | 
              </span>
              <span>
                DownVote
              </span>
            </button>
          </div>
          <div
            className="topic-item"
          >
            <h4>
              What is the best advice you can give to a junior programmer?
            </h4>
            <p>
              Oh I have a bunch of these.Been a Software Developer in multiple companies for close to 10 years and I could not stress more on the following
            </p>
            <button
              className="btn btn-default btn-sm btn-vote"
              onClick={[Function]}
            >
              <span>
                4
                 | 
              </span>
              <span>
                Vote
              </span>
            </button>
            <button
              className="btn btn-link btn-sm btn-downvote"
              onClick={[Function]}
            >
              <span>
                3
                 | 
              </span>
              <span>
                DownVote
              </span>
            </button>
          </div>
          <div
            className="topic-item"
          >
            <h4>
              How good of a programmer is Mark Zuckerberg?
            </h4>
            <p>
              In 2006, Zuckerberg gave up coding to focus on running the business of Facebook
            </p>
            <button
              className="btn btn-default btn-sm btn-vote"
              onClick={[Function]}
            >
              <span>
                15
                 | 
              </span>
              <span>
                Vote
              </span>
            </button>
            <button
              className="btn btn-link btn-sm btn-downvote"
              onClick={[Function]}
            >
              <span>
                5
                 | 
              </span>
              <span>
                DownVote
              </span>
            </button>
          </div>
          <div
            className="topic-item"
          >
            <h4>
              What are the best comments in a code?
            </h4>
            <p>
              Found at the beginning of an infinite loop
            </p>
            <button
              className="btn btn-default btn-sm btn-vote"
              onClick={[Function]}
            >
              <span>
                12
                 | 
              </span>
              <span>
                Vote
              </span>
            </button>
            <button
              className="btn btn-link btn-sm btn-downvote"
              onClick={[Function]}
            >
              <span>
                2
                 | 
              </span>
              <span>
                DownVote
              </span>
            </button>
          </div>
          <div
            className="topic-item"
          >
            <h4>
              What are some things that look easy but are difficult?
            </h4>
            <p>
              Doing stand up comedy. I do it as an amateur for fun.What I thought it was before I did it: You walk up, tell jokes for 10 minutes, then collect your free beer or $20.
            </p>
            <button
              className="btn btn-default btn-sm btn-vote"
              onClick={[Function]}
            >
              <span>
                12
                 | 
              </span>
              <span>
                Vote
              </span>
            </button>
            <button
              className="btn btn-link btn-sm btn-downvote"
              onClick={[Function]}
            >
              <span>
                3
                 | 
              </span>
              <span>
                DownVote
              </span>
            </button>
          </div>
          <div
            className="topic-item"
          >
            <h4>
              How India's Aadhaar database is maintained? What are the data structures involved? How search is performed (from an algorithmic perspective)?
            </h4>
            <p>
              
            </p>
            <button
              className="btn btn-default btn-sm btn-vote"
              onClick={[Function]}
            >
              <span>
                10
                 | 
              </span>
              <span>
                Vote
              </span>
            </button>
            <button
              className="btn btn-link btn-sm btn-downvote"
              onClick={[Function]}
            >
              <span>
                4
                 | 
              </span>
              <span>
                DownVote
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
